src/ToolBox/SOS/lldbplugin/CMakeLists.txt

```diff
@@ -81,12 +81,15 @@ find_path(LLDB_H "lldb/API/LLDB.h" PATHS "/usr/lib/llvm-3.8/include")
 find_path(LLDB_H "lldb/API/LLDB.h" PATHS "/usr/lib/llvm-3.7/include")
 find_path(LLDB_H "lldb/API/LLDB.h" PATHS "/usr/lib/llvm-3.6/include")
 find_path(LLDB_H "lldb/API/LLDB.h" PATHS "/usr/lib/llvm-3.5/include")
+#FreeBSD
+find_path(LLDB_H "lldb/API/LLDB.h" PATHS "/usr/local/llvm39/include")
+find_path(LLDB_H "lldb/API/LLDB.h" PATHS "/usr/local/llvm38/include")
 
 if(LLDB_H STREQUAL LLDB_H-NOTFOUND)
     if(REQUIRE_LLDBPLUGIN)
         message(FATAL_ERROR "Cannot find LLDB.h. Try installing lldb-3.6-dev (or the appropriate package for your platform)")
     else()
-        message(WARNING "Cannot find LLDB.h Try installing lldb-3.6-dev (or the appropriate package for your platform)")
+        message(WARNING "Cannot find LLDB.h Try installing lldb-3.6-dev (or the appropriate package for your platform and set LLVM_HOME)")
```

Comment:
Setting the LLVM_HOME is not needed in general, installing the lldb-dev package is sufficient on most platforms. Actually, I didn't need to do that on FreeBSD, just adding the find_path was sufficient.

Reply:
agreed. I wanted to improve message when normal lookup does not work.
Perhaps we can work on wording. In my case I have the headers but the build was not finding them. It may be easier IMHO for distro vendors to set variable instead maintain list of all possible locations.

Reply:
I would modify the wording to say:
Try installing lldb-3.6-dev (or the appropriate package for your platform). **You may need to set LLVM_HOME if the build still can't find it.**

I wonder - how did you install the LLVM stuff? I am asking since after we complete enabling FreeBSD, we should add instructions on what dependencies to install so that the build just works.

Reply:
@janvorli simple ` pkg install llvm38` should do the trick, also krb5 package is required at some point. I was following https://github.com/dotnet/coreclr/blob/master/Documentation/building/freebsd-instructions.md and it's 99% correct to the build point.